kie-soup-dataset/kie-soup-dataset-api/src/main/java/org/dashbuilder/dataset/json/DataSetMetadataJSONMarshaller.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dashbuilder.dataset.json;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.dashbuilder.dataset.ColumnType;
+import org.dashbuilder.dataset.DataSetMetadata;
+import org.dashbuilder.dataset.def.DataSetDef;
+import org.dashbuilder.dataset.impl.DataSetMetadataImpl;
+import org.dashbuilder.json.Json;
+import org.dashbuilder.json.JsonArray;
+import org.dashbuilder.json.JsonObject;
+
+public class DataSetMetadataJSONMarshaller {
+
+    public static final String UUID = "uuid";
+    public static final String NUMBER_OF_ROWS = "numberOfRows";
+    public static final String NUMBER_OF_COLUMNS = "numberOfColumns";
+    public static final String COLUMN_IDS = "columnIds";
+    public static final String COLUMN_TYPES = "columnTypes";
+    public static final String ESTIMATED_SIZE = "estimatedSize";
+    public static final String DEFINITION = "definition";
+
+    private DataSetDefJSONMarshaller dataSetDefJSONMarshaller;
+
+    public DataSetMetadataJSONMarshaller() {
+        // empty
+    }
+
+    public DataSetMetadataJSONMarshaller(DataSetDefJSONMarshaller dataSetDefJSONMarshaller) {
+        this.dataSetDefJSONMarshaller = dataSetDefJSONMarshaller;
+    }
+
+    public DataSetMetadata fromJSON(String json) {
+        JsonObject jsonObj = Json.parse(json);
+        return fromJsonObj(jsonObj);
+    }
+
+    public DataSetMetadata fromJsonObj(JsonObject json) {
+        JsonObject definitionObj = json.getObject(DEFINITION);
+        String uuid = json.getString(UUID);
+        int numberOfRows = json.getNumber(NUMBER_OF_ROWS).intValue();
+        int numberOfColumns = json.getNumber(NUMBER_OF_COLUMNS).intValue();
+        JsonArray columnIdsArray = json.getArray(COLUMN_IDS);
+        JsonArray columnTypesArray = json.getArray(COLUMN_TYPES);
+        int estimatedSize = json.getNumber(ESTIMATED_SIZE).intValue();
```

Comment:
Keep in mind that some of these `json.getXYZ()` methods can return `null`.
Is it something that needs to be handled?

Reply:
Hello @caponetto thanks for the comment. The `getNumber` method is safe, however, the `getArray` is indeed not safe, hence I added a check and a test to make sure that we avoid NPEs during marshalling!